src/js/videojs-dash.js

```diff
@@ -70,6 +70,21 @@ class Html5DashJS {
     // element to bind to.
     this.mediaPlayer_.initialize();
 
+    // Retrigger a dash.js-specific error event as a generic HTML5 one
+    // See src/streaming/utils/ErrorHandler.js in dash.js code
+    this.retriggerError_ = (event) => {
+      this.player.pause();
+      if (event.error === 'capability' && event.event === 'mediasource') {
+        this.player.error({code: 4});
+      } else if (event.error === 'capability' && event.event === 'encryptedmedia') {
+        this.player.error({code: 5});
+      } else if (event.error === 'download') {
```

Comment:
One thing we should be careful of is if the errors are fatal for dash.js. I'm not too familiar with their error system, but saw that dash.js' XHR module will trigger this download error if all retries fail for a specific request, but is it triggered for non-essential requests as well? For instance, if one rendition fails, will it trigger this error but continue to try other renditions?

Reply:
Ah jeez, came back to this PR after a while. Thought I'd gone through all the code paths making sure that was the case, looks like I didn't. Oops. Will finish this up today or tomorrow.